enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,413 @@
+and not in the ValidatingAdmissionWebhook.
+
+The controller's examination of the Validity condition in its existing Shared Resource reconciliation loop should 
+have minimal/negligible impact there.  And the controller's removal of data is a behavior whose path length is 
+already established in 4.10, as we remove data if the service account has its 'use' permissions on the Shared Resource
+revoked.
+
+#### Measuring / Verifying impact on existing API Server SLIs
+
+Existing metric discovery:
+- the EP template example in "Support Procedures" of using `webhook_admission_duration_seconds("<name of our new admission controller>", "validating", "put", "false")`, with >1s latency resulting in the `WebhookAdmissionLatencyHigh` alert firing, appears to be the valid choice addressing our new webhooks' impact on the API Object create/modify flows.  Is that correct?
+- Otherwise, items found from general k8s metric searches ....
+- the `apiserver_request_duration_seconds` is a histogram metric that has labels around resource/subresource, scope, and verb
+- the `kubelet_pod_start_duration_seconds_bucket` is a histogram metric that has labels around the create/update/sync operator types.
+
+Unclear how to sort the second and third metrics by namespace so that we could compare with Shared Resource metrics that could 
+convey "which namespaces are using Shared Resources and are slowing things down".  So perhaps they are not useful to us?
+
+
+#### Failure Modes
+
+1. Cluster Storage Operator / Shared Resource CSI Driver Operator failures could result in the ValidatingAdmissionWebhook not getting started
+2. API Server failures could result in the ValidatingAdmissionWebhook not getting called
+3. Unanticipated bugs in the ValidatingAdmissionWebhook result in incorrect validations
+
+Each of those ValidatingAdmissionWebhook checks either already are or will be made in the Shared Resource Controller and CSI Driver.
+So no violations of what we allow should occur, unless there are bugs across the entire path from webhook to controller to CSI driver for 
+a given constraint.
+
+Incorrectly preventing the create/update of Shared Resources at the moment appears to be relatively minor on the problem
+scale for cluster administrators and support.  But the error returned wil be directly available to the user.  That, along
+with the CLI invocation used, and any associated YAML for the object, should lead to straight forward diagnostics for 
+support.
+
+The enforcement of 'readOnly: true' in particular is worth discussing in this section.  It seems like very simple 
+pattern matching / field checking, but unanticipated bugs when performing this check
+in a ValidatingAdmissionWebhook could be impactful, as we are ultimately gating whether a Pod is allowed or not.  If 
+there is a bug which results in valid Pods being rejected, that could have broad and major consequences.
```

Comment:
If we are enforcing things at the `Pod` level, then ultimately every workload will run through this check.
This could potentially lead to undesirable behavior if the workload managing the pod does not gracefully handle repeated failed pod creation.

The same issue will apply to workload APIs that the webhook is not aware of. For instance, a Tekton TaskRun may create a faulty Pod and not report this error in a friendly manner.

Reply:
OK so I take it this comment is plus one'ing my concern @adambkaplan :-)

Your point about admission plugins vs. webhooks and getting back to the choice item that came up with the KEP are key.  Presumably with the reviewers here between you, me, @deads2k and @jsafrane will reach a final consensus.